### PR TITLE
Tweak captcha generation code

### LINKED ISSUE #2965: Captcha is unreadable
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.2.1_beta_2017-08-01
- **Operating system**: Fedora 27
- **Backend**: B2

## Description

I'm trying to delete a backup from the web interface but the captcha is unreadable as it is scaled in the webpage incorrectly.  The rest of the web interface is fine, nothing is unreadable.  Just the captcha:

image

Let me know what additional information, for example, from the browser would be helpful.

To note, I'm on a HiDPI screen with a resolution of 3840x2160 and 162 dpi.  I'm using Firefox 57.0.4.

## Steps to reproduce

1. Create a backup and backup some files
2. Try to delete the backup

### PR DESCRIPTION
**Pull request by oslerw**
Introduces a few changes to the code to generate captchas. Font size
calculations are now done in pixels for consistency, and the width of
the image is estimated from the width of the text being printed. Also
fixes #2965.

### CODE REVIEW

**Comment by oslerw** [comment]
Some examples of the new captchas:
new_captchas

File: Duplicati/Server/WebServer/CaptchaUtil.cs [hunk 1]
```diff
@@ -43,6 +43,16 @@ public static class CaptchaUtil
                 .Where(x => x != null)
                 .ToArray();
 
+        /// <summary>
+        /// Approximate the size in pixels of text drawn at the given fontsize
+        /// </summary>
+        private static int approxTextWidth(string text, FontFamily fontfamily, int fontsize) {
```
Comment (warwickmm) [inline Duplicati/Server/WebServer/CaptchaUtil.cs]: We usually begin method names with a capital letter.  Would you mind renaming this for consistency?

Also, the opening `{` typically appears on the next line.

**Comment by warwickmm** [comment]
These changes seem to work well for me.  I no longer (after around 50 attempts) encounter captchas that are obscured in the viewport. :+1:

**Comment by warwickmm** [comment]
Thanks!

**Pull request merged**